x-pack/plugin/watcher/src/main/java/org/elasticsearch/xpack/watcher/notification/email/Attachment.java

```diff
@@ -24,17 +24,25 @@
 import java.io.InputStream;
 import java.io.OutputStream;
 import java.nio.file.Path;
+import java.util.Collections;
+import java.util.Set;
 
 import static javax.mail.Part.ATTACHMENT;
 import static javax.mail.Part.INLINE;
 
 public abstract class Attachment extends BodyPartSource {
 
     private final boolean inline;
+    private final Set<String> warnings;
 
     protected Attachment(String id, String name, String contentType, boolean inline) {
```

Comment:
is this constructor used anymore? Im ok w/ removing it if we dont use it anywhere. Also, if we dont want a NPE down in the `EmailTemplate` we should check for null assignment in the other constructor, maybe with a `Objects.requireNonNull`. Otherwise the user is free to put null there and receive a nice NPE during `warnings.addAll(attachment.getWarnings());` below. Ill add a note with `here is the NPE im talking about`.

Reply:
Yeah, this constructor is still used in some cases. 

I added an `assert warnings != null`, since this is an internal API, the variable is final, means that someone working on Watcher internals would be required to explicitly call the second constructor with `null`, I think the assert is sufficient to express "don't do that".  https://github.com/elastic/elasticsearch/pull/44460/commits/ed5128fd3480ddde48b48974b337b6ac126fac0b